### PR TITLE
ADR-1908 Handle duplicate submission 422 response

### PR DESCRIPTION
**Pull request by matthew-goh**

### CODE REVIEW

File: app/uk/gov/hmrc/alcoholdutyreturns/connector/ReturnsConnector.scala [hunk 1]
```diff
@@ -91,29 +91,40 @@ class ReturnsConnector @Inject() (
...
             logger.warn(s"Not found returned for submit return (appaId $appaId, periodKey $periodKey)")
             Left(ErrorCodes.entityNotFound)
-          case Left(errorResponse)                                            =>
+          case response if response.status == UNPROCESSABLE_ENTITY =>
+            Try(response.json.as[DuplicateSubmissionError]) match {
+              case Success(dupError) if dupError.errors.code == "044" || dupError.errors.code == "999" =>
+                logger.info(s"Return already submitted (appaId $appaId, periodKey $periodKey)")
+                Left(ErrorCodes.duplicateSubmission)
+              case _                                                                                   =>
+                logger.warn(
```
Comment (ajchilts) [inline app/uk/gov/hmrc/alcoholdutyreturns/connector/ReturnsConnector.scala:120]: This is good as we can still monitor occurrences via the log entry. 👍

**Review by ajchilts — COMMENTED** [review]
@matthew-goh 

Have we tested the above solution in both ways? I.e. stubs returning the 422s as well as putting a sleep in stubs and letting the request timeout organically?

**Review by ajchilts — COMMENTED** [review]
Also did we test that the cache gets cleared the lock released when we process the duplicate 422s?

**Review by ajchilts — APPROVED** [review]
LGTM 👍

**Comment by matthew-goh** [comment]
Replies to the comments:
The cache does get cleared and the lock is released when processing the 422s.
I didn't add a case with the sleep to stubs, but I've tried it locally and the user goes to journey recovery without anything being cleared from Mongo. Noted that we don't have to add this timeout case to the stubs PR.
